Makefile

```diff
@@ -641,6 +659,8 @@ generate_yaml: $(HELM) $(HOME)/.helm
 		--namespace=istio-system \
 		--set global.hub=${HUB} \
 		--set global.proxy.enableCoreDump=${ENABLE_COREDUMP} \
+		--set istio_cni.enabled=${ENABLE_ISTIO_CNI} \
```

Comment:
if this is an environment variable, seems like the wrong way to do it.

Reply:
It's a Makefile var or env var, if set, that defaults to false.  It's the same pattern as ENABLE_COREDUMP.  

The modifications for generate_yaml are required to be compatible with e2e tests which rely on generate_yaml (istio.yaml and istio-auth.yaml).  That dependency is really not great and convoluted, IMO, but I'm not going to change all that e2e code.  So, the goal is to be able to use existing e2e test makefile targets with extra args to test with istio-cni enabled.